File: trace-agent-bootstrap/src/main/java/com/vmlens/trace/agent/bootstrap/interleave/lock/LockOperation.java
package com.vmlens.trace.agent.bootstrap.interleave.lock;

import com.vmlens.trace.agent.bootstrap.interleave.MonitorLockEnterStack;
import com.vmlens.trace.agent.bootstrap.util.IntStack;

public interface LockOperation {

	void execute(IntStack monitorStack ,IntStack exclusive , IntStack shared );

	void add2Stack(MonitorLockEnterStack monitorLockEnterStack, int threadIndex, int positionInThread);
	
	
}

File: sync-bug/src/main/scala/com/vmlens/preanalyzed/model/lockoperation/LockOperation.scala
package com.vmlens.preanalyzed.model.lockoperation

import com.vmlens.preanalyzed.model.LockType
import com.vmlens.trace.agent.bootstrap.preanalyzed.model.methodtypeimpl.AbstractMethodType

trait LockOperation {

  def create(lockType: LockType) : AbstractMethodType;

}

